packages/chevrotain/src/parse/grammar/lookahead.ts

```diff
@@ -441,8 +464,7 @@ class InsideDefinitionFinderVisitor extends GAstVisitor {
   ): void {
     if (
       node.idx === this.targetOccurrence &&
-      this.targetProdType === expectedProdName &&
-      (this.targetRef === undefined || node === this.targetRef)
```

Comment:
I am guessing this `targetRef` is some legacy code that is not used anywhere?

Reply:
Well, it was actually used in the `getLookaheadPathsForOr` function:

https://github.com/Chevrotain/chevrotain/blob/8096a8a23c0adb247913a6ad2219cf32a840d076/packages/chevrotain/src/parse/grammar/lookahead.ts#L613-L626

But it didn't really do anything, since the type of the production and it's occurrence have been enough to uniquely identify the target. As there's also no performance gain in the extra check, I've removed it.

Reply:
I looked at the history and it seems like this condition was introduced to handle
some edge case where there are duplicate occurrences
- https://github.com/Chevrotain/chevrotain/commit/33acea3814b35bd34481b4e2bf343c071c9a15e8